versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/ContentMapping.java

```diff
@@ -132,19 +132,17 @@ public Commit commitObjToCommit(
 
     if (fetchAdditionalInfo) {
       ContentKey key;
-      ContentMapping contentMapping = new ContentMapping(persist);
       IndexesLogic indexesLogic = indexesLogic(persist);
       for (StoreIndexElement<CommitOp> op : indexesLogic.commitOperations(commitObj)) {
         key = storeKeyToKey(op.key());
         // Note: key==null, if not the "main universe" or not a "content" discriminator
         if (key != null) {
           CommitOp c = op.content();
           if (c.action().exists()) {
-            commit.addOperations(
-                Put.of(
-                    key,
-                    contentMapping.fetchContent(
-                        requireNonNull(c.value(), "Required value pointer is null"))));
+            ObjId objId = requireNonNull(c.value(), "Required value pointer is null");
+            ContentValueObj contentValue =
+                persist.fetchTypedObj(objId, VALUE, ContentValueObj.class);
+            commit.addOperations(LazyPut.of(key, contentValue.payload(), contentValue.data()));
```

Comment:
I know, it's not changed here, but probably worth to first collect all `objId`s to load and then to a bulk-load instead of one `fetchTypedObj` for each `objId` individually.